clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -141,6 +141,11 @@ protected CompletableFuture<V> get(GetRequestContext requestContext, K key) thro
 
     CompletableFuture<V> valueFuture = new CompletableFuture<>();
     long timestampBeforeSendingRequest = System.nanoTime();
+
+    /**
```

Comment:
Check this:
https://github.com/linkedin/venice/blob/main/clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/LeastLoadedClientRoutingStrategy.java#L56

Reply:
That part of the code talks about the case: `requiredReplicaCount < availReplicas.size()`, my question was about the case: `requiredReplicaCount > availReplicas.size()`. I think these won't be an issue as we ideally don't set `requiredReplicaCount` more than 2 (original query + speculative query). But if 2 replicas out of 3 are down, this situation will be hit before helix comes in and brings up the 2nd and 3rd replicas.

Reply:
The request should fail if we don't have required replica count. If we are using speculativequery then we need to have 2 replicas available

Reply:
Speculative query should be best effort. If we have 1 replica available and speculative query enabled, then we should still execute that 1 request, not fail.

Reply:
Thanks @FelixGV. It's currently working that way and so resolving this.